internal/peer/lifecycle/chaincode/client_connections.go

```diff
@@ -118,17 +119,26 @@ func (c *ClientConnections) validatePeerConnectionParameters(input *ClientConnec
 		if err != nil {
 			return err
 		}
+		input.PeerAddresses = []string{}
+		input.TLSRootCertFiles = []string{}
+
+		if input.ChannelID == "" {
+			if input.TargetPeer == "" {
+				return errors.New("--targetPeer must be specified for channel-less operation using connection profile")
+			}
+			err := input.appendPeerConfig(networkConfig, input.TargetPeer)
+			if err != nil {
+				return err
+			}
+		}
+
```

Comment:
It seems to me this should return, `return input.appendPeerConfig(networkConfig, input.TargetPeer)`, since you've validated there is no ChannelID, the check below is never going to be `>0`

Reply:
Right, but it does need to go through the validation below that block. I should probably refactor all of this code based on the numerous things I've learned over the years about go programming but I'm not sure the effort is truly worth it for this CLI code.

Reply:
Totally missed expanding the code, and all of the validation logic below, sorry Will

Reply:
I'm refactoring a little though anyways. :)